src/sources/http_server.rs

```diff
@@ -568,15 +568,19 @@ impl HttpSource for SimpleHttpSource {
     /// This method adds the custom headers specified in the configuration
     /// to the HTTP response.
     fn enrich_reply<T: warp::Reply + 'static>(&self, reply: T) -> Box<dyn warp::Reply> {
-        let mut boxed_reply: Box<dyn warp::Reply> = Box::new(reply);
+        let mut response = reply.into_response();
+        let header_map = response.headers_mut();
+
         for (key, values) in &self.custom_response_headers {
-            boxed_reply = Box::new(warp::reply::with_header(
-                boxed_reply,
-                key,
-                values.join(", "),
-            ));
+            let header_name: HeaderName = key.parse().unwrap();
```

Comment:
Could we do this parsing and validation earlier? That is parse the configured headers into a `HeaderMap` during source start time? I think that would have the advantage of:

- Only parsing the headers once
- Being able to return an error at start time if a name is invalid, rather than panic'ing at runtime